### PR TITLE
994/review fixes

### PR DESCRIPTION
**Pull request by Heyrows**
## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/server/src/api/index.ts [hunk 1]
```diff
@@ -174,16 +174,16 @@ router.get(
 router.put(
   "/slices/rename",
   // eslint-disable-next-line @typescript-eslint/no-misused-promises
-  WithEnv(async function (
+  WithEnv(function (
```
Comment (mathieud) [inline packages/slice-machine/server/src/api/index.ts:177]: Curious about this change. What triggered it?

File: packages/slice-machine/server/src/api/push-changes.ts [hunk 2]
```diff
@@ -62,6 +58,47 @@ export default async function handler({
       body: null,
     };
 
+  // Fetch all models
+  const { localSlices, remoteSlices, localCustomTypes, remoteCustomTypes } =
```
Comment (mathieud) [inline packages/slice-machine/server/src/api/push-changes.ts]: This might be painful to merge with the screenshot changes :-(

**Review by mathieud — COMMENTED** [review]
Very cool! We can reconciliate with the screenshot changes this afternoon

**Review by mathieud — APPROVED** [review]

File: packages/slice-machine/components/DeleteDocumentsDrawer/DeleteDocumentsDrawerOverLimit.tsx [hunk 3]
```diff
@@ -38,6 +37,24 @@ const DeleteDocumentsDrawerOverLimit: React.FunctionComponent = () => {
...
 
+  const associatedDocumentsCards = modalData.details.customTypes.map(
+    (customTypeDetail) => {
+      const customType = availableCustomTypes.find(
+        (customType) => getModelId(customType) === customTypeDetail.id
+      );
+      if (customType === undefined || !isRemoteOnly(customType)) return null;
+
+      return (
+        <AssociatedDocumentsCard
```
Comment (danigo99) [inline packages/slice-machine/components/DeleteDocumentsDrawer/DeleteDocumentsDrawerOverLimit.tsx:48]: You're missing `isOverLimit` prop in here so they are styled differently

**Review by danigo99 — CHANGES_REQUESTED** [review]

File: packages/slice-machine/server/src/api/slices/save.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ import { SliceComparator } from "@prismicio/types-internal/lib/customtypes/diff"
 import { getOrElseW } from "fp-ts/lib/Either";
 import { SharedSlice } from "@prismicio/types-internal/lib/customtypes/widgets/slices";
 
-export async function handler(
+export function handler(
   env: BackendEnvironment,
   { sliceName, from, model: smModel, mockConfig }: SliceSaveBody
 ): Promise<Record<string, never>> {
```
Comment (danigo99) [inline packages/slice-machine/server/src/api/slices/save.ts]: We need to change this return type now also 🙂

File: packages/slice-machine/server/src/api/index.ts (hunk 1 above)
Comment (danigo99) [reply packages/slice-machine/server/src/api/index.ts:177]: A function call inside `generateLibraryIndex` was being awaited for no reason. And had an eslint disable `await-thenable`

**Review by danigo99 — APPROVED** [review]
I'm happy that all my comments were addressed. Looks clean 👌

File: packages/slice-machine/components/DeleteDocumentsDrawer/DeleteDocumentsDrawerOverLimit.tsx (hunk 3 above)
Comment (mathieud) [reply packages/slice-machine/components/DeleteDocumentsDrawer/DeleteDocumentsDrawerOverLimit.tsx:48]: How about this one?

**Review by mathieud — APPROVED** [review]

**Review by danigo99 — APPROVED** [review]

**Pull request merged**